packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -93,8 +140,7 @@ class WidgetFlutterBinding extends FlutterBinding {
 
 /// Inflate the given widget and attach it to the screen.
 void runApp(Widget app) {
-  WidgetFlutterBinding.ensureInitialized();
-  WidgetFlutterBinding.instance._runApp(app);
+  new WidgetFlutterBinding()._runApp(app);
```

Comment:
I guess the tricky part is it looks like you get a new binding every time you call runApp, which isn't really true...  Maybe we should just have an `ensureXXX` function that returns the instance and have runApp call that instead of calling the factory constructor.